File: TennisKata/Presenter.swift
class Presenter {
    private let model: Model
    var view: View? {
        didSet {
            view?.scoreDidChange(model.player1score.rawValue, model.player2score.rawValue)
        }
    }

    init(model: Model) {
        self.model = model
    }

    func player1() {
        let score = model.player1()
        let player1score = score.0
        let player2score = score.1

        view?.scoreDidChange(player1score.rawValue, player2score.rawValue)
    }

    func player2() {
        let score = model.player2()
        let player1score = score.0
        let player2score = score.1

        view?.scoreDidChange(player1score.rawValue, player2score.rawValue)
    }
}

File: android/app/src/main/java/io/github/gianpamx/tenniskatapresentationpatterns/mvp/Presenter.kt
package io.github.gianpamx.tenniskatapresentationpatterns.mvp

class Presenter(val view: View, val model: Model) {
    interface View {
        fun onScoreChange(player1Score: String, player2Score: String)
    }

    init {
        // Mapping (Map)
        val player1Score: String = model.player1score.name
        val player2Score: String = model.player2score.name

        view.onScoreChange(player1Score, player2Score)
    }

    fun player1() {
        val scorePair = model.player1()

        // Mapping (Map)
        val player1Score: String = scorePair.first.name
        val player2Score: String = scorePair.second.name

        view.onScoreChange(player1Score, player2Score)
    }

    fun player2() {
        val scorePair = model.player2()

        // Mapping (Map)
        val player1Score: String = scorePair.first.name
        val player2Score: String = scorePair.second.name

        view.onScoreChange(player1Score, player2Score)
    }
}
